pkg/sql/scheduledlogging/captured_index_usage_stats_test.go

```diff
@@ -157,18 +154,20 @@ func TestCaptureIndexUsageStats(t *testing.T) {
 	expectedTotalNumEntriesInSingleInterval := 8
 	expectedNumberOfIndividualIndexEntriesInSingleInterval := 1
 
-	// Expect index usage statistics logs once the schedule disabled interval has passed.
-	// Assert that we have the expected number of total logs and expected number
-	// of logs for each index.
-	testutils.SucceedsWithin(t, func() error {
-		return checkNumTotalEntriesAndNumIndexEntries(t,
-			expectedIndexNames,
-			expectedTotalNumEntriesInSingleInterval,
-			expectedNumberOfIndividualIndexEntriesInSingleInterval,
-		)
-	}, stubScheduleCheckEnabledInterval+timeBuffer)
-
-	// Verify that a second schedule has run after the enabled interval has passed.
+	// Wait for channel value from end of 1st schedule.
+	<-scheduleCompleteChan
+	time.Sleep(timeBuffer)
```

Comment:
Why do you need this? Can't you have the main code signal to the test when it's finished writing to logs?

Reply:
https://github.com/cockroachdb/cockroach/pull/89319/files#diff-567a23559822ff60b8d9d63b7fda0e82a78d11fa6dced531e3f6d24966282362R165

The main code signals to the channel here, after the logging has completed here: https://github.com/cockroachdb/cockroach/pull/89319/files#diff-567a23559822ff60b8d9d63b7fda0e82a78d11fa6dced531e3f6d24966282362R153.

 For some reason - that I have not been able to figure out why - the logs don't appear if you check immediately, despite logging being completed. The 1 second buffer seems to consistently allow for the logs to appear in the log file, including under stress.

Reply:
maybe you're missing `log.Flush`?

Reply:
Yup, was missing `log.Flush`, will remove the `time.Sleep` calls.